nns/nns_contract.go

```diff
@@ -260,6 +260,12 @@ func parentExpired(ctx storage.Context, first int, fragments []string) bool {
 }
 
 // Register registers a new domain with the specified owner and name if it's available.
+//
+// Access rules::
+//   - TLD can be registered only by the committee
+//   - 2nd-level domain can be registered by anyone
+//   - starting from the 3rd level, the domain can only be registered by the
+//     owner or administrator (if any) of the previous level domain
```

Comment:
BTW, current implementation doesn't support more than 3 levels.

Reply:
didn't pay attention..
is this an intentional/desired limitation? What does C# say?

Reply:
I doubt it's intentional, do we have an issue for it?

Reply:
@AnnaShaleva where exactly is this limitation? I didn’t find it from the sources, the test with L4 also passes

Reply:
found that this has already been changed f12c44fc786a9ed4e6ff24a38be3cde7cd624ee9